src/openms/include/OpenMS/SYSTEM/BuildInfo.h

```diff
@@ -0,0 +1,225 @@
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Julianus Pfeuffer $
+// $Authors: Julianus Pfeuffer $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/build_config.h>
+#include <OpenMS/DATASTRUCTURES/String.h>
+#include <QSysInfo>
+#include <QString>
+#ifdef _OPENMP
+  #include "omp.h"
+#endif
+
+namespace OpenMS
+{
+  namespace Internal
+  {
+
+    enum OpenMS_OS {OS_UNKNOWN, OS_MACOS, OS_WINDOWS, OS_LINUX};
+    std::string OpenMS_OSNames[] = {"unknown", "MacOS", "Windows", "Linux"};
+    enum OpenMS_Architecture {ARCH_UNKNOWN, ARCH_32BIT, ARCH_64BIT};
+    std::string OpenMS_ArchNames[] = {"unknown", "32 bit", "64 bit"};
+
+    #if WIN32
+    OpenMS_Architecture getArchOnWin();
+    String getWinOSVersion();
+    #endif
+
+    class OpenMSOSInfo
+    {
+      OpenMS_OS os_;
+      String os_version_;
+      OpenMS_Architecture arch_;
+
+    public:
+      OpenMSOSInfo() :
+          os_(OS_UNKNOWN),
+          os_version_("unknown"),
+          arch_(ARCH_UNKNOWN)
+      {}
+
+      /// @brief Get the current operating system (Windows, MacOS, Linux)
+      String getOSAsString() const
+      {
+        return OpenMS_OSNames[os_];
+      }
+
+      /// @brief Get the current architecture (32-bit or 64-bit)
+      String getArchAsString() const
+      {
+        return OpenMS_ArchNames[arch_];
+      }
+
+      /// @brief Get the OS version (e.g. 10.15 for macOS or 10 for Windows)
+      String getOSVersionAsString() const
+      {
+        return os_version_;
+      }
+
+      /// @brief Get Architecture of this binary (simply by looking at size of a pointer, i.e. size_t).
+      static String getBinaryArchitecture()
+      {
+        size_t bytes = sizeof(size_t);
+        switch (bytes)
+        {
+          case 4:
+            return OpenMS_ArchNames[ARCH_32BIT];
+          case 8:
+            return OpenMS_ArchNames[ARCH_64BIT];
+          default:
+            return OpenMS_ArchNames[ARCH_UNKNOWN];
+        }
+      }
+
+      /// @brief Constructs and returns an OpenMSOSInfo object
+      static OpenMSOSInfo getOSInfo()
+      {
+        OpenMSOSInfo info;
+        #if defined(WIN32)  // Windows
+        info.os_ = OS_WINDOWS;
+        info.arch_ = getArchOnWin();
+        info.os_version_ = getWinOSVersion();
+        #elif (defined(__MACH__) && defined(__APPLE__)) // MacOS
+        info.os_ = OS_MACOS;
+        #else //Linux
+        info.os_ = OS_LINUX;
+        #endif
+        // check if we can use QSysInfo
+        #if (defined(Q_OS_MACOS) || defined(Q_OS_UNIX))
+        info.os_version_ = QSysInfo::productVersion();
+        // identify architecture
+        if (QSysInfo::WordSize == 32)
```

Comment:
so for non-win, this is always identical to pointer-size, but for Win, this could be larger (due to wow64)?
However, we do not do 32bit builds anymore... so this will never be used, right?

Reply:
I'd say, just report the  OpenMS binary architecture by looking at sizeof(size_t), and ignore the system arch in all cases (because that should not be our concern).

Reply:
ok done. I agree